tools/public_api_guard/material/bottom-sheet.d.ts

```diff
@@ -20,7 +20,7 @@ export declare const matBottomSheetAnimations: {
 
 export declare class MatBottomSheetConfig<D = any> {
     ariaLabel?: string | null;
-    autoFocus?: boolean;
+    autoFocus?: string;
```

Comment:
Additionally, the new type wouldn't be `string`, but instead include the enum type:
```ts
autoFocus?: AutoFocusTarget | string | boolean;
```
I know that AutoFocusTarget is a subtype of string, but adding the enum should improve IDE autocomplete

The removed version would also be v14 at this point

Reply:
I ended up adding the comment in these .d.ts files. Although in the code I mainly saw them in .ts files. Is there any convention on where they should go?

Reply:
The comment should go in the actual source; these `.d.ts` files in `public_api_guard` are automatically generated, so any changes there will be clobbered.

For context: when a TypeScript file is compiled, it generates the `.js` output file and a `.d.ts` type definition file. The type definition is produced so that other TypeScript code can type check against it without having to compile _everything_ all together. We take of the type definition file with our public_api_guards to make sure we don't make unintentional changes to the public api. There's a test that compares the existing public api `.d.ts` file to the one from the PR and fails if it changed, so that any intentional changes have to be explicitly approved.